app/controllers/application_controller.rb

```diff
@@ -137,11 +143,24 @@ def authenticate_token
       @session = Session.find(token)
       return false if @session.nil?
       @current_user = User.find(@session.uuid)
-      extend_session!
-      return true if @current_user.present?
+      if should_signout_sso?
+        destroy_user_session!(@current_user, @session)
```

Comment:
The flow of this is confusing to me.. I think the return value of this method (authenticate_token) now relies on MHVLoggingService.logout's return value in the destroy_user_session! method .. might be a good place for an explicit return to make it clear what the intended behavior is

Reply:
actually, it returns on the `ensure` portion of the destroy_user_session block, which would be whether or not curren_user, session, etc were able to be destroyed, but that's a fair point. If for some reason those raise an exception we might have a problem, but I can't think of a better way of handling those than what we're already doing in that method.

Reply:
I spent a fair amount of time this morning thinking the same thing -- that it would return what's in ensure, but it surprised me that it doesn't seem to be the case :) that destroy logic in ensure would return truthy anyways so we still wouldn't want that
irb example:
```
dhall$ irb
2.3.0 :001 > def try_me
2.3.0 :002?>   puts 'hi'
2.3.0 :003?>   true
2.3.0 :004?>   ensure
2.3.0 :005 >     false
2.3.0 :006?>   end
 => :try_me
2.3.0 :007 > try_me
hi
 => true
```

Reply:
ohhhhhhhhh wow!

Reply:
I removed MHVLoggingService.logout(user) entirely, lets hope specs still pass. But I think this should remove the ambiguity.